src/v/cluster/topic_table.h

```diff
@@ -397,6 +430,15 @@ class topic_table {
     /// Returns metadata of all topics.
     const underlying_t& all_topics_metadata() const;
 
+    std::optional<partition_replicas_info>
```

Comment:
we established a new code documentation policy, please add a comment to public methods 
(more inforation here: https://github.com/redpanda-data/redpanda/pull/16400/commits/47deba02d9a3d1d0298b196f52e769beaa153130#diff-0168c31ef7ebccbf9f8b89d28b02f6ab8647e956ad737a0c7ad9f6ab7161eba7)